server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala

```diff
@@ -183,8 +183,12 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
       @CacheEntryExpired
       def onCacheEvent(event: CacheEntryEvent[Bytes, Bytes]) {
          if (isSendEvent(event)) {
-            val dataVersion = event.getMetadata.version().asInstanceOf[NumericVersion].getVersion
-            sendEvent(event.getKey, event.getValue, dataVersion, event)
+            sendEvent(event.getKey, event.getValue, if (event.getMetadata == null) {
```

Comment:
Minor: the 3rd param to ```sendEvent``` could be written in a more functional way as:

```
Option(event.getMetadata)
      .map(_.version().asInstanceOf[NumericVersion].getVersion)
      .getOrElse(null.asInstanceOf[Long])
```

It'd be nicer without the ugly casts, but this cast is around in several places. It assumes ```NumericVersion``` is always used, so I wonder if there's a valid use case for setting SIMPLE versioning scheme on the server. If so, it'd break in several places. Not related to this PR anyway.